### PR TITLE
Elm tests use Data module for identifiers

### PR DESCRIPTION
**Pull request by aoldershaw**
## Changes proposed by this PR:

**tl;dr** this only cleans up tests - when we implement pipeline instances in the UI, we'll need to make changes to all of these tests anyway (else we'll encounter syntax errors)

Adding new fields to identifiers is a pain (as you can probably tell from this PR) because it introduces compiler errors in probably >100 tests.  Rather than each test defining its own identifiers, use default ones + helper functions (e.g. `withName "new-value"`) that transform the default values.

The motivation for this is to prepare for pipeline instances which will introduce a new parameter for identifying pipelines (and therefore also jobs, resources, and resource versions). With this change, when we add a new field to the identifier records, we'll only need to change the default values in the `Data` module.

## Contributor Checklist

- [x] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [x] [Signed] all commits
- [x] Added tests (Unit and/or Integration)
- [ ] Updated [Documentation]
- [ ] Updated [Release notes]

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs
[Release notes]: https://github.com/concourse/concourse/tree/master/release-notes

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

File: web/elm/tests/BuildTests.elm [hunk 1]
```diff
@@ -101,18 +97,7 @@ all =
...
-                            , nextBuild = Nothing
-                            , finishedBuild = Nothing
-                            , transitionBuild = Nothing
-                            , paused = False
-                            , disableManualTrigger = False
-                            , inputs = []
-                            , outputs = []
-                            , groups = []
-                            }
+                            (Data.job 0 |> Data.withShortPipelineId |> Data.withName "j")
```
Comment (jomsie) [inline web/elm/tests/BuildTests.elm:100]: neat!
